File: frontend/src/app/entity/antwort.ts
import {Test} from "./test";
import {Frage} from "./frage";

export interface Antwort {
  test?: Test,
  question?: Frage,
  id: number,
  beschreibung: string,
  korrekt: boolean,
  korrekterAntwortText: string
}

export const ANSWERTYPE_MULTIPLECHOICE = 0;
export const ANSWERTYPE_TEXT = 1;

File: backend/src/main/java/com/example/azubiquiz/entity/Antwort.java
package com.example.azubiquiz.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import lombok.*;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@ToString
@Entity
@Table(name = "antwort")
public class Antwort {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id", nullable = false)
    private long id;

    @NotBlank
    private String beschreibung;

    private boolean korrekt;

    private String korrekterAntwortText;

    @ManyToOne
    @JoinColumn(name = "frage_id")
    @JsonIgnore
    private Frage frage;

}